src/main/resources/assets/src/managerView/branchHoursView.js

```diff
@@ -1,13 +1,283 @@
+import $ from 'jquery';
 import React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { EditDialog, EditorData, EditItem } from './editDialog';
+import { getTopResults, convertTime24to12, convertTime12to24 } from '../functions';
+import HeaderFilters from './headerFilters';
 
 export default class BranchHoursView extends React.Component {
     constructor(props) {
         super(props);
+        this.state = {
+            showEditDialog: false,
+            editorData: undefined,
+            filters: [[], [], [], []],
+        }
+    }
+
+    editBranchHours(index, id, ...items) {
+        items.pop();
+        this.setState({
+            showEditDialog: true,
+            editorData: new EditorData(id, items, [], index * 35 + 87),
+        })
+    }
+
+    addBranchHours(editItems) {
+        this.setState({
+            showEditDialog: true,
+            editorData: new EditorData(-1, editItems, [], this.props.branchHours.length * 35 + 93),
+        })
+    }
+
+    async deleteBranchHours(id) {
+        this.props.setStateValue('loading', true);
+
+        await $.ajax({
+            type: 'DELETE',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            url: '/api/branchHours/delete/' + id + '/'
+        });
+
+        await this.props.loadData('branchHours');
+        this.props.setStateValue('loading', false);
+    }
+
+    async saveEdits(id, newValues) {
+        this.props.setStateValue('loading', true);
+
+        const openTime = convertTime12to24(newValues[0]);
+        const closeTime = convertTime12to24(newValues[1]);
+        const branch = this.props.branches.find(b => b.name == newValues[2]);
+        const dayOfWeek = this.convertDayToInt(newValues[3]);
+
+        const validOpenTime = openTime.match('^(0[0-9]|1[0-9]|2[0-3]|[0-9]):[0-5][0-9]$');
+        const validCloseTime = closeTime.match('^(0[0-9]|1[0-9]|2[0-3]|[0-9]):[0-5][0-9]$');
+
+        // Check if the data is valid
+        if (!validOpenTime || !validCloseTime || !branch || !dayOfWeek) {
+            const newEditorData = this.state.editorData;
+
+            if (!validOpenTime) {
+                newEditorData[0] = "Invalid open time";
```

Comment:
These should be: `newEditorData.editErrors[0]`. I realized I messed that up too for some of them.

Reply:
Got you. Update these now.